xdsl/dialects/arm/assembly.py

```diff
@@ -21,10 +21,15 @@ class reg(AssemblyInstructionArg):
     """
 
     value: SSAValue
+    is_ptr_to_mem: bool | None = False
 
-    def __init__(self, value: SSAValue) -> None:
+    def __init__(self, value: SSAValue, *, is_ptr_to_mem: bool = False) -> None:
         self.value = value
+        self.is_ptr_to_mem = is_ptr_to_mem
 
     def assembly_str(self) -> str:
         assert isinstance(self.value.type, RegisterType)
-        return self.value.type.register_name.data
+        if self.is_ptr_to_mem:
+            return f"[{self.value.type.register_name.data}]"
+        else:
+            return self.value.type.register_name.data
```

Comment:
Can you please put this in a new class instead? Maybe named `PointerReferenceArg`? Not sure that's the best name.

Reply:
Or maybe just `square_brackets_reg`?

Reply:
I've updated it to ```square_brackets_reg```